### ISSUE TITLE
Limit on number of steps

### ISSUE THREAD

**Issue by bschulzke** [bjfoutz00/thesaurus-game#2]
Does your function which calls to the thesaurus API check the minimum number of steps between two words before offering the challenge? I haven't thoroughly inspected your code, but it seems like this would be a useful feature if you haven't added it, allowing you to filter out pairings that aren't within 10 or 20 steps. You could also optionally display to the user the minimum path between a pair. This would reassure players that there really is a path in every case.

**Comment by jaredblack** [comment]
We've thought about this and it would be a really good feature to have, unfortunately it would be very costly in terms of API calls to traverse all (or even just some) paths between words to check if it's possible. If we had our own thesaurus database that we were using, it would be more feasible. Right now, all of the words in the bank so far have had a path between them, usually within less than 10 which gives me confidence that virtually all pairings of common words should have some sort of path between them.

**Comment by bschulzke** [comment]
Fair enough! Another feature that might be easier than that to implement, which everyone I've talked to has agreed would improve their experience, is a back button to go back one step at a time, rather than just a restart. As it stands, everyone I've shared the game with so far has essentially done this by restarting and then retracing their steps their steps each time they want to go back a bit, but it is more tedious and frustrating. Since that's how most people seem to be playing anyway, seems like a back button wouldn't fundamentally change the game, it would just smooth over the user experience.
